lib/language_pack/cache.rb

```diff
@@ -69,4 +102,9 @@ def cache_copy(from,to)
   def exists?(path)
     File.exists?(@cache_base + path)
   end
+
+  # Returns a hash of a directory's recursive filenames and mtimes.
+  def dir_hash(path)
+    `ls -laAgGR --time-style=+%s #{path} | awk '{print $4 $5}' | sha1sum | head -c 40`
```

Comment:
You should just be able to use `mtime` directly, on Linux (and probably mac) the `mtime` of a directory will get updated if any files are changed (i.e. if any inodes are added or removed).

```suggestion
    `stat -c %Y node_modules`
```

Reply:
That'd mean we'd have to be OK with not catching files modified in place, which is why I originally went with `du` instead. I guess we could do a combination of the two? :/

Reply:
Oh, and mtime changing when inodes are added/removed is not recursive, it turns out. Only the mtime of the file's direct parent.

Reply:
Apparently, this also seems to work
```suggestion
    File.mtime(path)
```

Reply:
This is where I landed after fiddling with it some more:

`find #{path} -printf "%T@" | sha1sum`

`%T` is mtime for the `find` command, and this had the same-ish performance as `du`, so I'm happy with this for recursively detecting if a directory has changed.